src/system_data.jl

```diff
@@ -378,6 +400,8 @@ function transform_single_time_series!(
                 interval,
             )
             check_add_time_series!(data.time_series_params, params)
+            !_is_uninitialized(data.time_series_params.forecast_params) &&
```

Comment:
This check is system-wide but conducted inside the for loop for every component.